### ISSUE TITLE
How to import Exo player libray or aar file to Android AOSP

### ISSUE THREAD

**Issue by jagdish-dev** [google/ExoPlayer#10403]
This Android Open Source Project (AOSP) repository offers the information and source code needed to create custom variants of the Android OS, port devices, and accessories for the Android platform.

The ExoPlayer library is located in the external folder of the Android AOSP framework. 
**Here, while integrating the existing ExoPlayer to any class it produces the resources import errors.**
**package com.google.android.exoplayer2.ui does not exist
import com.google.android.exoplayer2.ui.PlayerView;**

We have integrated the ExoPlayer to Android.bp file also. As following code,

java_import {
    name: "exoplayer-2.15.0-sources",
    host_supported: true,
    installable: false,
    jars: ["exoplayer-2.15.0-sources.jar"],
    }

java_import {
    name: "exoplayer-core-2.15.0-sources",
    host_supported: true,
    installable: false,
    jars: ["exoplayer-core-2.15.0-sources.jar"],
    }

java_import {
    name: "exoplayer-dash-2.15.0-sources",
    host_supported: true,
    installable: false,
    jars: ["exoplayer-dash-2.15.0-sources.jar"],
    }

java_import {
    name: "exoplayer-hls-2.15.0-sources",
    host_supported: true,
    installable: false,
    jars: ["exoplayer-hls-2.15.0-sources.jar"],
    }

java_import {
    name: "exoplayer-smoothstreaming-2.15.0-sources",
    host_supported: true,
    installable: false,
    jars: ["exoplayer-smoothstreaming-2.15.0-sources.jar"],
    }

java_import {
    name: "exoplayer-ui-2.15.0-sources",
    host_supported: true,
    installable: false,
    jars: ["exoplayer-ui-2.15.0-sources.jar"],
    }
    
    
    Please suggest resolving this import issue and how to integrate ExoPlayer to AOSP.

**Comment by marcbaechinger** [comment]
> The ExoPlayer library is located in the external folder of the Android AOSP framework.

ExoPlayer as the library here in this repository providing classes in the packages `com.google.android.exoplayer2.*` is kind of legacy already in favour of Media3 that you can find here in a new module and package structure: https://github.com/androidx/media

Both are external libraries with Media3 being part of the AndroidX/Jetpack libraries. The source code is hosted on GitHub in this repo and the Media3 repo from above. Binaries are distributed by Google Maven. Release notes can be found [here](https://developer.android.com/jetpack/androidx/releases/media3) with a note about hosting on Google Maven.

Can you give me some context around why do you think that ExoPlayer should be part of the AOSP repository?

**Comment by jagdish-dev** [comment]
Hello Marc,
Thank you for the feedback.

We have used com.google.android.exoplayer2.* in the Android Application and it is working perfectly fine to play the video.
We have a requirement to integrate ExoPlayer in Android System AOSP. White adding this library to **Android. bp** file, it is unable to access in class. We have used the below-mentioned statements in Android. bp.

java_import {
name: "exoplayer-2.15.0-sources",
host_supported: true,
installable: false,
jars: ["exoplayer-2.15.0-sources.jar"],
}  

Now adding these imports in class file throws an exception and not accesible.
**import com.google.android.exoplayer2.ExoPlayerFactory;
import com.google.android.exoplayer2.Player;
import com.google.android.exoplayer2.SimpleExoPlayer;**

Is there any documentation available to integrate the Exoplayer in Android AOSP? It would be helpful.
